### PR TITLE
Fix/pierreh/fix todos

### PR DESCRIPTION
**Pull request by phieronymus**
No ticket associated to this PR.

The only TODO remaining is the test todo for form data. If handled it would be in another Pull Request as investigation is needed. Other todos/fixmes have been fixed in this Pull Request

### CODE REVIEW

File: src/errors/RequestResponseError.ts [hunk 1]
```diff
@@ -4,50 +4,81 @@
...
-        super(serializeErrorResponse(errorResponse));
+    constructor(errorResponse: RawErrorRequest) {
+        const formattedErrorResponse = formatErrors(errorResponse);
+        super(serializeErrorResponse(formattedErrorResponse));
+
         Object.setPrototypeOf(this, RequestResponseBaseError.prototype);
-        this.errorResponse = { status: "error", ...errorResponse };
+        this.errorResponse = { status: "error", ...formattedErrorResponse };
     }
 }
```
Comment (phieronymus) [inline src/errors/RequestResponseError.ts:71]: Ok, wha changed here is that the error is formatted to fit the right type. Before the `serializeErrorResponse` was handling this but the error inside the `RequestResponseBaseError` still was ill-formatted that can cause issues in the consumers when using the object as the type was wrong.

Now it should be consistent with the information typescript is giving.

File: src/utils/object.ts [hunk 2]
```diff
@@ -66,9 +66,3 @@ export function missingKeys(obj: Record<string, any>, keys: string[] = []): stri
 
     return missing;
 }
-
-export function omit<T extends Record<string, unknown>, K extends keyof T>(obj: T, keys: K[]): Omit<T, K> {
```
Comment (phieronymus) [inline src/utils/object.ts:70]: This function was not needed anymore.

File: src/errors/RequestResponseError.ts (hunk 1 above)
Comment (phieronymus) [reply src/errors/RequestResponseError.ts:71]: Note that more tests have been added here as well.

File: test/specs/errors.specs.ts [hunk 3]
```diff
@@ -33,4 +33,51 @@ describe("Errors", () => {
             "Error: Code: REQUEST_ERROR, HttpCode: 400, Errors: REQUEST_ERROR, REQUEST_ERROR (my no reason field)"
         );
     });
+
+    it("should serialize the ResponseError correctly", async () => {
```
Comment (laxa88) [inline test/specs/errors.specs.ts:37]: This description is duplicated. 🤔

File: test/specs/errors.specs.ts [hunk 4]
```diff
@@ -33,4 +33,51 @@ describe("Errors", () => {
...
+                { reason: ResponseErrorCode.InvalidFormat, field: "dummy_field" },
+            ],
+        });
+
+        expect(error.toString()).to.equal(
+            "Error: Code: VALIDATION_ERROR, HttpCode: 400, Errors: INVALID_AMOUNT, INVALID_FORMAT (dummy_field)"
+        );
+    });
+
+    it("should serialize an error with `errors` not being an array", async () => {
```
Comment (laxa88) [inline test/specs/errors.specs.ts:52]: probably can shorten this to `should serialize non-array errors`

File: test/specs/errors.specs.ts [hunk 5]
```diff
@@ -33,4 +33,51 @@ describe("Errors", () => {
...
+        const error = new RequestError({
+            httpCode: 400,
+            code: RequestErrorCode.RequestError,
+            errors: { reason: RequestErrorCode.RequestError },
+        });
+
+        expect(error.toString()).to.equal("Error: Code: REQUEST_ERROR, HttpCode: 400, Errors: REQUEST_ERROR");
+    });
+
+    it("should serialize an error with `errors` using a symbol as error item", async () => {
```
Comment (laxa88) [inline test/specs/errors.specs.ts:62]: probably can shorten this to `should serialize symbol errors`

File: test/specs/errors.specs.ts [hunk 6]
```diff
@@ -33,4 +33,51 @@ describe("Errors", () => {
...
+            code: RequestErrorCode.RequestError,
+            errors: [RequestErrorCode.RequestError, "invalid code", 1, true],
+        });
+
+        expect(error.toString()).to.equal(
+            "Error: Code: REQUEST_ERROR, HttpCode: 400, Errors: REQUEST_ERROR, UNKNOWN_ERROR, UNKNOWN_ERROR, UNKNOWN_ERROR"
+        );
+    });
+
+    it("should serialize an error with `errors` having null and undefined values", async () => {
```
Comment (laxa88) [inline test/specs/errors.specs.ts:74]: probably can shorten this to `should serialize undefined and null errors`

**Review by laxa88 — APPROVED** [review]

File: test/specs/errors.specs.ts (hunk 4 above)
Comment (phieronymus) [reply test/specs/errors.specs.ts:52]: Actually it would be confusing as the `error` object contains the `errors` object/array. Not specifying will make it harder to know which `error` we speak about 🤔

Comment (phieronymus) [reply test/specs/errors.specs.ts:52]: Same for the two next comments. I believe that shortening it will introduce some understanding issue as the object is structured in a way which is prone to misunderstanding when describing.

I would feel better being a bit verbose to describe exactly what we want. But I may be missing something here.

File: test/specs/errors.specs.ts (hunk 3 above)
Comment (phieronymus) [reply test/specs/errors.specs.ts:37]: Ups. Seems that we already had the test. Thanks 🙂
